src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -280,37 +280,167 @@ def parse_known_args(self, args=None, namespace=None):
         self._namespace, self._raw_arguments = super().parse_known_args(args=args, namespace=namespace)
         return self._namespace, self._raw_arguments
 
-    def _check_value(self, action, value):
+    def _get_extension_command_tree(self):
+        from azure.cli.core._session import EXT_CMD_TREE
+        import os
+        VALID_SECOND = 3600 * 24 * 10
+        # self.cli_ctx is None when self.prog is beyond 'az', such as 'az iot'.
+        # use cli_ctx from cli_help which is not lost.
+        cli_ctx = self.cli_ctx or (self.cli_help.cli_ctx if self.cli_help else None)
+        if not cli_ctx:
+            return None
+        EXT_CMD_TREE.load(os.path.join(cli_ctx.config.config_dir, 'extensionCommandTree.json'), VALID_SECOND)
+        if not EXT_CMD_TREE.data:
+            import requests
+            from azure.cli.core.util import should_disable_connection_verify
+            try:
+                response = requests.get(
+                    'https://azurecliextensionsync.blob.core.windows.net/cmd-index/extensionCommandTree.json',
+                    verify=(not should_disable_connection_verify()),
+                    timeout=300)
+            except Exception as ex:  # pylint: disable=broad-except
+                logger.info("Request failed for extension command tree: %s", str(ex))
+                return None
+            if response.status_code == 200:
+                EXT_CMD_TREE.data = response.json()
+                EXT_CMD_TREE.save_with_retry()
+            else:
+                logger.info("Error when retrieving extension command tree. Response code: %s", response.status_code)
+                return None
+        return EXT_CMD_TREE
+
+    def _search_in_extension_commands(self, command_str):
+        """Search the command in an extension commands dict which mimics a prefix tree.
+        If the value of the dict item is a string, then the key represents the end of a complete command
+        and the value is the name of the extension that the command belongs to.
+        An example of the dict read from extensionCommandTree.json:
+        {
+            "aks": {
+                "create": "aks-preview",
+                "update": "aks-preview",
+                "app": {
+                    "up": "deploy-to-azure"
+                },
+                "use-dev-spaces": "dev-spaces"
+            },
+            ...
+        }
+        """
+
+        cmd_chain = self._get_extension_command_tree()
+        for part in command_str.split():
+            try:
+                if isinstance(cmd_chain[part], str):
+                    return cmd_chain[part]
+                cmd_chain = cmd_chain[part]
+            except KeyError:
+                return None
+        return None
+
+    def _get_extension_use_dynamic_install_config(self):
+        cli_ctx = self.cli_ctx or (self.cli_help.cli_ctx if self.cli_help else None)
```

Comment:
cli_ctx seems could be `None` and then line 342 would crash.

Reply:
line 342 has the logic `if cli_ctx else 'no'` to handle `None`, it will return `None`.